### DISCUSSION TITLE
Index deployment on the Production environment ?

### DISCUSSION THREAD

**Discussion by mcrio** [ravendb/ravendb#19018] [General]
Regarding the production index deployment best practices described here:
https://ravendb.net/articles/managing-ravendb-indexes-in-production-a-devops-guide

I'm still wondering how people do it the most safe way?

Having a specific endpoint which deploys the index from code requires you to have the upcoming release on the production server, with the endpoint accessible but the release not yet live. Another way is to have a separate binary that does the deployment, and again it has to be part of the production network. It's doable but not that straight forward, and it also doesn't include auto-indexes, and I'd rather have more control and deploy the indexes one by one manually.

The other solution is to export the index and import to the production database.

That's the most easy way, but it can easily lead to a disaster, as when making the DB export you need to deselect everything that was pre-selected automatically and choose only the indexes. You need to verify like 50 times that you haven't selected the documents and other irrelevant non production data. So this way of updating indexes can be nerve wracking.

Possible Solution:

I wish the RavenDB Studio would have a simple "Export" button for each Index on that page. Optimally it would make an export into a file with a custom extension like "ravenindexdump". Than there would be another "Import" button on that page that would allow importing that "ravenindexdump" export.

**Comment by ppekrol** [comment]
hi @mcrio 

You are correct, we want to make it more straightforward for our users as well. This is why that feature is under the development: https://issues.hibernatingrhinos.com/issue/RavenDB-18066/Consider-adding-a-quicker-way-to-copy-single-and-multiple-indexes-between-databases
We are now wrapping up 6.1, expect us to get back to this issue soon.
